gslib/boto_translation.py

```diff
@@ -651,13 +652,14 @@ def _PerformSimpleDownload(self,
                              num_progress_callbacks=XML_PROGRESS_CALLBACKS,
                              headers=None,
                              hash_algs=None):
-    try:
+    if isinstance(key, GsKey):
```

Comment:
I feel we should ask more clarifying questions to the user as I do not see any reason why the existing code would not catch TypeError.

Reply:
Plus, we are not even sure that this fix is going to resolve the issue. It might be some other issue like I have mentioned offline in the chat.